### PR TITLE
Added GitHub action with welcome message for first-time contributors.

### PR DESCRIPTION
**Pull request by orf**
See [here for context](https://forum.djangoproject.com/t/improving-the-contribution-experience-with-github-actions/5964/32).

The exact wording probably needs refining, but the jist of the implementation is there.

### CODE REVIEW

File: .github/workflows/new_contributor_pr.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+on:
+  pull_request_target:
+
+jobs:
+  build:
+    name: Hello new contributor
+    runs-on: ubuntu-latest
+    steps:
+      # Pinned to v2.0
+      - uses: deborah-digges/new-pull-request-comment-action@224c179a9e23f65ec50ff3240b8716369dc415d7
```
Comment (carltongibson) [inline .github/workflows/new_contributor_pr.yml:12]: This action limits to first contribution yes? 🙇‍♂️🏃

**Review by carltongibson — APPROVED** [review]
Hey @orf — I like this. We should probably give it a run as it is. Thanks! 

I'm +1 here — let me just confirm with @felixxm

File: .github/workflows/new_contributor_pr.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          message: |
+            Hello ${{ github.event.pull_request.user.login }}! Thank you for your contribution 💪
+
+            As it's your first contribution be sure to check out the [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist).
+            If you're fixing a ticket [from Trac](code.djangoproject.com/) make sure to set the "Has Patch" property
+            and include a link to this PR in the ticket!
+
+            If you have any design or process questions then you can ask in the [Django Development forums](https://forum.djangoproject.com/c/internals/5).
+
+            Thanks again, you rock! 🚀
```
Comment (carltongibson) [inline .github/workflows/new_contributor_pr.yml:24]: We often use `Welcome aboard ⛵️` — maybe just that.

File: .github/workflows/new_contributor_pr.yml (hunk 1 above)
Comment (orf) [reply .github/workflows/new_contributor_pr.yml:12]: Yep - it lists PRs within the repository by that user: https://github.com/Deborah-Digges/new-pull-request-comment-action/blob/main/src/index.js#L9-L15.

It's probably simple enough to inline with a script [like I did here](https://github.com/django/django/pull/13961/files#diff-721c8b3a3c86e303692c0fd6f744bb92e76eaf44146145c9baa8b65d7099b9d4R33), but for now 🤷

File: .github/workflows/new_contributor_pr.yml (hunk 2 above)
Comment (orf) [reply .github/workflows/new_contributor_pr.yml]: Done 👍

File: .github/workflows/new_contributor_pr.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+      # Pinned to v2.0
+      - uses: deborah-digges/new-pull-request-comment-action@224c179a9e23f65ec50ff3240b8716369dc415d7
+        with:
+          access-token: ${{ secrets.GITHUB_TOKEN }}
+          message: |
+            Hello ${{ github.event.pull_request.user.login }}! Thank you for your contribution 💪
+
+            As it's your first contribution be sure to check out the [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist).
+            If you're fixing a ticket [from Trac](code.djangoproject.com/) make sure to set the "Has Patch" property
```
Comment (felixxm) [inline .github/workflows/new_contributor_pr.yml:19]:
```suggestion
            If you're fixing a ticket [from Trac](https://code.djangoproject.com/) make sure to set the _"Has Patch"_ property
```

File: .github/workflows/new_contributor_pr.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  build:
+    name: Hello new contributor
+    runs-on: ubuntu-latest
+    steps:
+      # Pinned to v2.0
+      - uses: deborah-digges/new-pull-request-comment-action@224c179a9e23f65ec50ff3240b8716369dc415d7
+        with:
+          access-token: ${{ secrets.GITHUB_TOKEN }}
+          message: |
+            Hello ${{ github.event.pull_request.user.login }}! Thank you for your contribution 💪
```
Comment (felixxm) [inline .github/workflows/new_contributor_pr.yml:16]: Should we add `@`?
```suggestion
            Hello @${{ github.event.pull_request.user.login }}! Thank you for your contribution 💪
```

File: .github/workflows/new_contributor_pr.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+            As it's your first contribution be sure to check out the [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist).
```
Comment (felixxm) [inline .github/workflows/new_contributor_pr.yml:18]: I would add a blank line:
```suggestion
            As it's your first contribution be sure to check out the [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist).
            
```

File: .github/workflows/new_contributor_pr.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            If you're fixing a ticket [from Trac](code.djangoproject.com/) make sure to set the "Has Patch" property
+            and include a link to this PR in the ticket!
+
+            If you have any design or process questions then you can ask in the [Django Development forums](https://forum.djangoproject.com/c/internals/5).
```
Comment (felixxm) [inline .github/workflows/new_contributor_pr.yml:22]:
```suggestion
            If you have any design or process questions then you can ask in the [Django forum](https://forum.djangoproject.com/c/internals/5).
```

**Review by felixxm — COMMENTED** [review]
@orf Thanks :+1:

Personally, I don't like automatic messages but I'm also not against them. Feel-free to merge this.

File: .github/workflows/new_contributor_pr.yml (hunk 4 above)
Comment (orf) [reply .github/workflows/new_contributor_pr.yml:16]: Yeah makes sense!

File: .github/workflows/new_contributor_pr.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  build:
+    name: Hello new contributor
+    runs-on: ubuntu-latest
+    steps:
+      # Pinned to v2.0
+      - uses: deborah-digges/new-pull-request-comment-action@224c179a9e23f65ec50ff3240b8716369dc415d7
+        with:
+          access-token: ${{ secrets.GITHUB_TOKEN }}
+          message: |
+            Hello @${{ github.event.pull_request.user.login }}! Thank you for your contribution 💪
```
Comment (felixxm) [inline .github/workflows/new_contributor_pr.yml:16]: We can use `{}`:
```suggestion
            Hello @{}! Thank you for your contribution 💪
```

**Review by felixxm — APPROVED** [review]
@orf Thanks :+1: I pushed minor edits.

We tested this message in https://github.com/felixxm/django/pull/2 (Thanks @MarkusH :star:).

**Comment by MarkusH** [comment]
:heart:  But can we sneak in a :unicorn: emoji in the message?

**Comment by felixxm** [comment]
> heart But can we sneak in a unicorn emoji in the message?

I like :unicorn: but I think the welcome message it's fine without it :wink:

**Pull request merged**